### PR TITLE
Add V3 for XSwap Protocol

### PR DESCRIPTION
**Pull request by pro100skm**
1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Please enable "Allow edits by maintainers" while putting up the PR.
3. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
4. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
5. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
6. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
7. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
xspswap (fork of Uniswap V2) already added into DefiLlama [here](https://github.com/DefiLlama/DefiLlama-Adapters/pull/4107)

Current updates adding V3 and a few core assets of XDC Network.

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/xspswap-v3 exports TVL: 
        
 
 
    xdc                       42.62 k
    
    total                    42.62 k

File: projects/xspswap-v3/index.js
Comment (g1nt0ki) [inline projects/xspswap-v3/index.js:1]: this looks good

File: projects/xspswap-v2/index.js
Comment (g1nt0ki) [inline projects/xspswap-v2/index.js:1]: renaming folder breaks our server, we will handle the name change ourselves :)

File: projects/helper/coreAssets.json
Comment (g1nt0ki) [inline projects/helper/coreAssets.json:1]: rejecting this change as I try to keep this list as small as possible

**Comment by g1nt0ki** [comment]
@pro100skm thanks for the PR, included your changes here: https://github.com/DefiLlama/DefiLlama-Adapters/commit/3549c1dc47b0e45fbd6c587491c07bb0ecf8fdf2

File: projects/xspswap-v2/index.js
Comment (pro100skm) [reply projects/xspswap-v2/index.js:1]: ok, thank you

File: projects/helper/coreAssets.json
Comment (pro100skm) [reply projects/helper/coreAssets.json:1]: NP

**Pull request closed without merging**